src/test/utils/VaultUtils.sol

```diff
@@ -1,152 +1,453 @@
+    // or state variable tests should be tested in the inherited contract
+    UniswapVault public vault;
 
-    function setUp() public virtual override {
+    function setUp() public override {
         super.setUp();
+        createTokensAndDexPair();
+        vault = UniswapVault(payable(deployVault()));
+    }
+
+    /////////////////////////// VIRTUAL FUNCTIONS ///////////////////////////////
+
+    // can choose weth as base or some arbitrary token
+    function createTokensAndDexPair() internal virtual {}
+
+    // deploys the vault with chosen parameters
+    function deployVault() internal virtual returns (address) {}
+
+    // allows tests to acquire tokens for various purposes, different if it's a simple erc20 or WETH
+    function getToken0(address _user, uint256 _amount) internal virtual;
+
+    function getToken1(address _user, uint256 _amount) internal virtual;
 
-        token0 = new TestToken("Test Token 0", "TT0");
-        token1 = new TestToken("Test Token 1", "TT1");
-
-        factory.createPair(address(token0), address(token1));
-
-        uniswapVaultImpl = new UniswapVault();
-        uniswapVault = UniswapVault(
-            payable(
-                address(
-                    new TransparentUpgradeableProxy(
-                        address(uniswapVaultImpl),
-                        address(proxyAdmin),
-                        abi.encodeWithSignature(
-                            "initialize(address,uint256,address,address,uint256,uint256,address,address)",
-                            address(core),
-                            1,
-                            address(token0),
-                            address(token1),
-                            10_000,
-                            500,
-                            address(factory),
-                            address(router)
-                        )
-                    )
-                )
-            )
-        );
+    ///////////////////////////// BASIC TESTS ////////////////////////////////////
+    /// These tests should pass regardless of vault type and parameters chosen ///
+    function test_initParams() public {
+        assertEq(address(pair), factory.getPair(address(token0), address(token1)));
+        assertTrue(vault.isNativeVault() == (address(token0) == C.WETH));
     }
-}
 
-contract MasterChefVaultFixture is UniswapFixture {
-    TestToken public token0;
-    TestToken public token1;
-    MasterChefMock public masterChef;
-    MasterChefVault public masterChefVaultImpl;
-    MasterChefVault public masterChefVault;
+    function test_depositZeroAmountRevert() public {
+        vm.expectRevert("ZERO_AMOUNT");
+        vault.depositToken0(0);
+    }
 
-    function setUp() public virtual override {
-        super.setUp();
+    function test_depositToken0WithValueRevert() public {
+        if (!vault.isNativeVault()) {
+            vm.expectRevert("NOT_NATIVE_VAULT");
+            vault.depositToken0{ value: 1 }(0);
+        }
```

Comment:
what is the expected behavior for `depositToken0(<some non zero value>)` on a native vault

Reply:
fair point, it ignores that value and is not expected to revert. I'll rename to be more accurate